File: app1/app/src/main/java/com/example/app1/MainActivity.java
package com.example.app1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Parcelable;
import android.view.View;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void submit(View view){
        EditText etName = findViewById(R.id.etName);
        EditText etAge = findViewById(R.id.etAge);
        RadioGroup rg = findViewById(R.id.rbSex);

        String name = etName.getText().toString();
        String age = etAge.getText().toString();

        if(name.equals("") || age.equals("") || rg.getCheckedRadioButtonId() == -1)
            Toast.makeText(this,"Введите все необходимые данные",Toast.LENGTH_SHORT).show();
        else {
            int rbId = rg.getCheckedRadioButtonId();
            RadioButton rb = findViewById(rbId);
            String sex = rb.getText().toString();

            Intent i = new Intent(MainActivity.this, SecondActivity.class);
            i.putExtra("iName", name);
            i.putExtra("iAge", age);
            i.putExtra("iSex", sex);
            startActivity(i);
        }
    }
    public void clear(View view){
        EditText name = findViewById(R.id.etName);
        EditText age = findViewById(R.id.etAge);
        RadioGroup rg = findViewById(R.id.rbSex);
        name.setText("");
        age.setText("");
        rg.clearCheck();
    }
}

File: app5/app/src/main/java/com/example/app5/MainActivity.kt
package com.example.app5

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ProgressBar
import android.widget.Toast
import com.google.firebase.database.FirebaseDatabase

class MainActivity : AppCompatActivity() {
    private val db = FirebaseDatabase.getInstance().getReference("User")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        findViewById<Button>(R.id.loginBTNSignin).setOnClickListener{
            var login = findViewById<EditText>(R.id.loginETLogin)
            var password = findViewById<EditText>(R.id.loginETPassword)
            db.child(login.text.toString()).get().addOnSuccessListener{
                if(it.exists()){
                    if(password.text.toString() == it.child("pwd").value){
                        val i = Intent(this@MainActivity, MainActivity3::class.java)
                        i.putExtra("user", login.text.toString())
                        startActivity(i)
                        Toast.makeText(this, "Добро пожаловать!", Toast.LENGTH_SHORT).show()
                    }else Toast.makeText(this, "Пароль неверный", Toast.LENGTH_SHORT).show()
                }else Toast.makeText(this, "Логин неверный", Toast.LENGTH_SHORT).show()
            }
        }
        findViewById<Button>(R.id.loginBTNSignup).setOnClickListener{
            val i = Intent(this@MainActivity, MainActivity2::class.java)
            startActivity(i)
        }
    }
}